packages/dashboard/src/managers/trajectory-socket-manager.ts

```diff
@@ -32,7 +32,7 @@ export default class TrajectorySocketManager extends EventEmitter<Events> {
     if (!webSocket) throw Error('Websocket not initialized!');
     // response should come in the order that requests are sent, this should allow multiple messages
     // in-flight while processing the responses in the order they are sent.
-    webSocket.send(payload);
+    if (webSocket.readyState === WebSocket.OPEN) webSocket.send(payload);
```

Comment:
wouldn't this cause an infinite wait as no request is actually sent? You probably should throw an error in this case.

Reply:
I thought the connection to the backend is made here?

```
<UserContext.Provider value={user}>
    <TrajectorySocketContext.Provider value={new WebSocket(appConfig.trajServerUrl)}>
        <ThemeProvider theme={theme}>
```

Once the connection to the backend is successful, then the websocket's ready state will be marked as `open` and the request would be sent.

Anyway, I also realized that switching between the tabs causes multiple client connections to be made to the dashboard, make it very laggy after several switches. I'm also going to solve this problem in this pr too

Reply:
This PR adds a condition so that it only sends a request if the websocket is opened, if it is not, it continues execution without sending anything, so then later in this part

```tsx
    this._ongoingRequest = new Promise((res) => {
      this._listenOnce('message', webSocket, (e) => {
        this._ongoingRequest = null;
        res(e);
      });
    });
```

We create a promise that is resolved when we get a `message` event. since we never sent any requests, it's possible that there will never be a response so the promise would never get resolved.